### PR TITLE
Add basic completions for shallow-backup

### PR DESCRIPTION
**Pull request by alichtman**
This adds basic tab completion for
https://github.com/alichtman/shallow-backup/

This will fix https://github.com/alichtman/shallow-backup/issues/324.

I am the original author of this piece of software.

- [x] This compdef is not already available in zsh.
- [x] This compdef is not already available in their original project.
- [x] I am the original author, or I have authorization to submit this work.
- [x] This is a finished work.
- [x] It has a header containing authors, status and origin of the script.
- [x] It has a license header or I accept that it will be licensed under the terms of the Zsh license.

### CODE REVIEW

File: src/_shallow-backup [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#
+#  * Aaron Lichtman (https://github.com/alichtman)
+#
+# ------------------------------------------------------------------------------
+
+
+_shallow-backup() {
+    _arguments \
+      '(- 1 *)'-h, --help'[Show help and exit]' \
+      '(- 1 *)'-v, --version'[Print version]' \
```
Comment (syohex) [inline src/_shallow-backup:45]:
```suggestion
      '(- 1 *)'{-h,--help}'[Show help and exit]' \
      '(- 1 *)'{-v,--version}'[Print version]' \
```

File: src/_shallow-backup [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      '--add-dot[Add a dotfile or dotfolder to config by path]':PATH \
```
Comment (syohex) [inline src/_shallow-backup:46]:
```suggestion
      '--add-dot[Add a dotfile or dotfolder to config by path]:PATH:_files' \
```

If you want to complete files or directories, you can use `_files` action

File: src/_shallow-backup [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      '--backup-all[Full back up]' \
+      '--backup-configs[Back up app config files]' \
+      '--backup-dots[Back up dotfiles]' \
+      '--backup-fonts[Back up installed fonts]' \
+      '--backup-packages[Back up package libraries]' \
+      '--delete-config[Delete config file]' \
+      '--destroy-backup[Delete backup directory]' \
+      '--dry-run[Do not backup or reinstall any files, just give verbose output]' \
+      '--new-path[Input a new back up directory path]':PATH \
```
Comment (syohex) [inline src/_shallow-backup:55]:
```suggestion
      '--new-path[Input a new back up directory path]:PATH:_files -/' \
```

Ditto. We can use `_files -/` to complete only directories

File: src/_shallow-backup [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      '--no-new-backup-path-prompt[Skip setting new back up directory path prompt]' \
+      '--no-splash[Do not display splash screen]' \
+      '--reinstall-all[Full reinstallation]' \
+      '--reinstall-configs[Reinstall configs]' \
+      '--reinstall-dots[Reinstall dotfiles and dotfolders]' \
+      '--reinstall-fonts[Reinstall fonts]' \
+      '--reinstall-packages[Reinstall packages]' \
+      '--remote[Set remote URL for the git repo]':url \
+      '--show[Display config file]' \
```
Comment (syohex) [inline src/_shallow-backup:64]:
```suggestion
      '--show[Display config file]'
```

File: src/_shallow-backup [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+_shallow-backup
```
Comment (syohex) [inline src/_shallow-backup:67]:
```suggestion
_shallow-backup

# Local Variables:
# mode: Shell-Script
# sh-indentation: 2
# indent-tabs-mode: nil
# sh-basic-offset: 2
# End:
# vim: ft=zsh sw=2 ts=2 et
```

**Comment by alichtman** [comment]
Thanks! @syohex

**Pull request merged**